backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/FilterPrivilegedDatasetsHandler.java

```diff
@@ -98,50 +100,49 @@ public FilterPrivilegedDatasetsHandler(Executor executor, FutureJdbi jdbi) {
+              List<InternalFuture<Optional<String>>> internalFutures = new LinkedList<>();
               for (String datasetId : datasetIds) {
-                try {
-                  internalFutures.add(
-                      InternalFuture.completedInternalFuture(datasetId)
-                          .thenCompose(
-                              id ->
-                                  permissionCheck
-                                      .getEntityPermissionBasedOnResourceTypes(
-                                          Collections.singletonList(datasetId),
-                                          ModelDBActionEnum.ModelDBServiceActions.READ,
-                                          ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET)
-                                      .thenAccept(
-                                          allowed -> {
-                                            if (!allowed) {
-                                              throw new PermissionDeniedException(
-                                                  "Permission denied");
-                                            }
-                                          },
-                                          executor)
-                                      .thenCompose(
-                                          unused -> InternalFuture.completedInternalFuture(id),
-                                          executor),
-                              executor));
-
-                } catch (PermissionDeniedException ex) {
-                  LOGGER.trace(ex.getMessage());
-                  // We are throws error while creating entities
-                  // if we are fetching entities then we are just skip those datasets from response
-                  // and skipped PermissionDeniedException on fetch
-                  if (errorOut) {
-                    return InternalFuture.failedStage(ex);
-                  }
-                }
+                internalFutures.add(
+                    InternalFuture.completedInternalFuture(datasetId)
+                        .thenCompose(
+                            id ->
+                                permissionCheck
+                                    .getEntityPermissionBasedOnResourceTypes(
+                                        Collections.singletonList(datasetId),
+                                        ModelDBActionEnum.ModelDBServiceActions.READ,
+                                        ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET)
+                                    .thenApply(
+                                        allowed -> {
+                                          // We are throws error while creating entities
+                                          // if we are fetching entities then we are just skip those
+                                          // datasets from response
+                                          // and skipped PermissionDeniedException on fetch
```

Comment:
Asked about this on slack as I don't follow

Reply:
Update the comment